plugins/tiledb/io/TileDBWriter.cpp

```diff
@@ -0,0 +1,345 @@
+    for (const auto& kv : mapAttrVals)
+    {
+        Dimension::Id id = Dimension::id(kv.first);
+        Dimension::Type t = view->dimType(id);
+        switch (t)
+        {
+            case Dimension::Type::Double:
+            {
+                auto sp = pdalboost::any_cast<std::shared_ptr<std::vector<double>>>(kv.second);
+                m_query->set_buffer(kv.first, *sp);
+                break;
+            }
+            case Dimension::Type::Float:
+            {
+                auto sp = pdalboost::any_cast<std::shared_ptr<std::vector<float>>>(kv.second);
+                m_query->set_buffer(kv.first, *sp);
+                break;
+            }
+            case Dimension::Type::Signed8:
+            {
+                auto sp = pdalboost::any_cast<std::shared_ptr<std::vector<char>>>(kv.second);
+                m_query->set_buffer(kv.first, *sp);
+                break;
+            }
+            case Dimension::Type::Signed16:
+            {
+                auto sp = pdalboost::any_cast<std::shared_ptr<std::vector<short>>>(kv.second);
+                m_query->set_buffer(kv.first, *sp);
+                break;
+            }
+            case Dimension::Type::Signed32:
+            {
+                auto sp = pdalboost::any_cast<std::shared_ptr<std::vector<int>>>(kv.second);
+                m_query->set_buffer(kv.first, *sp);
+                break;
+            }
+            case Dimension::Type::Signed64:
+            {
+                auto sp = pdalboost::any_cast<std::shared_ptr<std::vector<long>>>(kv.second);
+                m_query->set_buffer(kv.first, *sp);
+                break;
+            }
+            case Dimension::Type::Unsigned8:
+            {
+                auto sp = pdalboost::any_cast<std::shared_ptr<std::vector<unsigned char>>>(kv.second);
+                m_query->set_buffer(kv.first, *sp);
+                break;
+            }
+            case Dimension::Type::Unsigned16:
+            {
+                auto sp = pdalboost::any_cast<std::shared_ptr<std::vector<unsigned short>>>(kv.second);
+                m_query->set_buffer(kv.first, *sp);
+                break;
+            }
+            case Dimension::Type::Unsigned32:
+            {
+                auto sp = pdalboost::any_cast<std::shared_ptr<std::vector<unsigned int>>>(kv.second);
+                m_query->set_buffer(kv.first, *sp);
+                break;
+            }
+            case Dimension::Type::Unsigned64:
+            {
+                auto sp = pdalboost::any_cast<std::shared_ptr<std::vector<unsigned long>>>(kv.second);
+                m_query->set_buffer(kv.first, *sp);
+                break;
+            }
+            case Dimension::Type::None:
+            default:
+                throw pdal_error("Unsupported attribute type for " + kv.first);
+
+        }
+    }
+
+    m_query->set_coordinates(coords);
+    m_query->submit();
+}
+
+void TileDBWriter::done(PointTableRef table)
+{
+    m_query->finalize();
```

Comment:
Finalize is only strictly needed for global order writes.

Reply:
:+1: